libazureinit/src/provision/password.rs

```diff
@@ -25,18 +25,34 @@ impl PasswordProvisioner {
 fn passwd(user: &User) -> Result<(), Error> {
     // Update the sshd configuration to allow password authentication.
     let sshd_config_path = "/etc/ssh/sshd_config.d/50-azure-init.conf";
-    let ret = update_sshd_config(sshd_config_path);
-    if ret.is_err() {
+    if let Err(error) = update_sshd_config(sshd_config_path) {
+        tracing::error!(
+            ?error,
+            sshd_config_path,
+            "Failed to update sshd configuration for password authentication"
+        );
         return Err(Error::UpdateSshdConfig);
     }
     let path_passwd = env!("PATH_PASSWD");
 
     if user.password.is_none() {
-        let status = Command::new(path_passwd)
-            .arg("-d")
-            .arg(&user.name)
-            .status()?;
+        let mut command = Command::new(path_passwd);
+        command.arg("-d").arg(&user.name);
+        tracing::trace!(?command, "About to remove user's password");
```

Comment:
One thing worth noting is this would log the username (if TRACE logging is enabled, which it wouldn't be by default)